### PR TITLE
Attach a uuid to each event so dedupe can happen server-side

### PR DESCRIPTION
**Pull request by grahamc**
##### Checklist

- [ ] Tested changes against a test repository
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] (If this PR is for a release) Updated README to point to the new tag (leave unchecked if not applicable)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,41 +10,37 @@
     "check-fmt": "prettier --check .",
     "format": "prettier --write .",
     "lint": "eslint src/**/*.ts",
-    "prebuild": "cp package.json ./src/package.json",
+    "prebuild": "echo 'export const pkg =' > src/package.ts; cat package.json >> ./src/package.ts && prettier --write ./src/package.ts && eslint --fix ./src/package.ts",
```
Comment (lucperkins) [inline package.json:13]: Could you clarify why this is here? Feels quite icky, and just want to make sure it's truly necessary.

**Review by lucperkins — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -106,4 +106,4 @@ lib/**/*
 # Nix builds
 result
 
-src/package.json
+src/package.ts
```
Comment (cole-h) [inline .gitignore:109]: ...this seems wrong?

Comment (cole-h) [reply .gitignore:109]: (There is no `src/package.json` _or_ `src/package.ts` file)

Comment (cole-h) [reply .gitignore:109]: Oh, I see, it's related to Luc's comment above...

File: package.json (hunk 1 above)
Comment (grahamc) [reply package.json:13]: I want to embed the version from the package.json in the metrics being submitted without maintaining it in two places. This seems to work fine as package.json when depended upon in other actions, but when I run run-test.js after minification, it doesn't work with complaints that I need need to import like this: `import info from `./package.json` assert { type: 'json' };` or like this: `import info from `./package.json` with { type: 'json' };`, and that causes other issues.

File: .gitignore (hunk 2 above)
Comment (grahamc) [reply .gitignore:109]: ya b/c it's .gitignored :)

File: package.json (hunk 1 above)
Comment (grahamc) [reply package.json]: Made much nicer in https://github.com/DeterminateSystems/detsys-ts/pull/16/commits/c65adbc917507d589245819a34b3983236aa5027

**Review by lucperkins — APPROVED** [review]

File: src/correlation.ts [hunk 3]
```diff
@@ -4,17 +4,17 @@ import { createHash } from "node:crypto";
...
-  run_differentiator?: string;
+  runDifferentiator?: string;
   workflow?: string;
   groups: Record<string, string | undefined>;
 };
 
 export function identify(projectName: string): AnonymizedCorrelationHashes {
   const ident = {
-    correlation_source: "github-actions",
+    correlationSource: "github-actions",
```
Comment (grahamc) [inline src/correlation.ts:17]: these need to be _'d because the server expects them to be _'d

Comment (cole-h) [reply src/correlation.ts:17]: (fwiw, you can use serde to add aliases so that both `correlation_source` and `correlationSource` are accepted)

Comment (lucperkins) [reply src/correlation.ts]: Open to suggestion but I'm going to use snake case here and attach a comment (and also turn on camel case checking in ES Lint, plus appropriate ignore statements).

**Pull request merged**